### PR TITLE
Created KITTI dataset for segmentation in autonomous driving scenario

### PR DESCRIPTION
**Pull request by TimoK93**
Note that this PR is a modified version of the withdrawn PR https://github.com/open-mmlab/mmsegmentation/pull/1748

## Motivation

In the last years, panoptic segmentation has become more into the focus in reseach. Weber et al. [[Link]](http://www.cvlibs.net/publications/Weber2021NEURIPSDATA.pdf) have published a quite nice dataset, which is in the same style like Cityscapes, but for KITTI sequences. Since Cityscapes and KITTI-STEP share the same classes and also a comparable domain (dashcam view), interesting investigations, e.g. about relations in the domain e.t.c. can be done. 

Note that KITTI-STEP provices panoptic segmentation annotations which are out of scope for mmsegmentation.

## Modification

Mostly, I added the new dataset and dataset preparation file. To simplify the first usage of the new dataset, I also added configs for the dataset, segformer and deeplabv3plus.  

## BC-breaking (Optional)

No BC-breaking

## Use cases (Optional)

Researchers want to test their new methods, e.g. for interpretable AI in the context of semantic segmentation. They want to show, that their method is reproducible on comparable datasets. Thus, they can compare Cityscapes and KITTI-STEP.

### CODE REVIEW

**Comment by MeowZheng** [comment]
thanks again for your contribution. we are working on reviewing it.

File: configs/_base_/datasets/kittistep.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    dict(type='Normalize', **img_norm_cfg),
+    dict(type='Pad', size=crop_size, pad_val=0, seg_pad_val=255),
+    dict(type='DefaultFormatBundle'),
+    dict(type='Collect', keys=['img', 'gt_semantic_seg']),
+]
+test_pipeline = [
+    dict(type='LoadImageFromFile'),
+    dict(
+        type='MultiScaleFlipAug',
+        img_scale=None,
```
Comment (csatsurnh) [inline configs/_base_/datasets/kittistep.py:23]:
```suggestion
        img_scale=(1242, 375),
```

File: projects/kitti_step_dataset/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+| --------- | -------- | --------- | ------: | -------- | -------------- | ----: | ------------: | ---------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Segformer | MIT-B5   | 368x368   |  160000 | -        | -              | 65.05 |             - | [config](configs/segformer/segformer_mit-b5_368x368_160k_kittistep.py) | [model](https://openmmlab-share.oss-cn-hangzhou.aliyuncs.com/mmsegmentation/v0.5/segformer/segformer_mit-b5_368x368_160k_kittistep/segformer_mit-b5_368x368_160k_kittistep_20230506_103002-20797496.pth) | [log](https://openmmlab-share.oss-cn-hangzhou.aliyuncs.com/mmsegmentation/v0.5/segformer/segformer_mit-b5_368x368_160k_kittistep/segformer_mit-b5_368x368_160k_kittistep_20230506_103002.log.json) |
```
Comment (xiexinch) [inline projects/kitti_step_dataset/README.md:45]: Might use 'https://download.openmmlab/com/xxx'

File: projects/kitti_step_dataset/mmseg/datasets/__init__.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                                      LoadImageNetSImageFromFile)
+from mmseg.datasets.isaid import iSAIDDataset
+from mmseg.datasets.isprs import ISPRSDataset
+from mmseg.datasets.loveda import LoveDADataset
+from mmseg.datasets.night_driving import NightDrivingDataset
+from mmseg.datasets.pascal_context import (PascalContextDataset,
+                                           PascalContextDataset59)
+from mmseg.datasets.potsdam import PotsdamDataset
+from mmseg.datasets.stare import STAREDataset
+from mmseg.datasets.voc import PascalVOCDataset
```
Comment (xiexinch) [inline projects/kitti_step_dataset/mmseg/datasets/__init__.py:27]: We don't need to import all dataset classes.

File: projects/kitti_step_dataset/mmseg/datasets/kitti_step.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+from mmseg.datasets.builder import DATASETS
+from mmseg.datasets.custom import CustomDataset
+
+
+@DATASETS.register_module()
+class KITTISTEPDataset(CustomDataset):
+    """KITTI-STEP dataset.
+
+    This class is adapted from ./cityscapes.py
```
Comment (xiexinch) [inline projects/kitti_step_dataset/mmseg/datasets/kitti_step.py:16]:
```suggestion
    This class is adapted from cityscapes
```

File: projects/kitti_step_dataset/docs/en/dataset_prepare.md [hunk 5]
```diff
@@ -0,0 +1,663 @@
...
+│   │   │   ├── train-semi-segmentation
+│   │   │   ├── validation
+│   │   │   ├── validation-segmentation
+│   │   │   ├── test
+│   │   ├── ImageNetS50
+│   │   │   ├── train-semi
+│   │   │   ├── train-semi-segmentation
+│   │   │   ├── validation
+│   │   │   ├── validation-segmentation
+│   │   │   ├── test
```
Comment (xiexinch) [inline projects/kitti_step_dataset/docs/en/dataset_prepare.md:176]: I think don't need this file and we might move the preparation steps to README.md.

File: projects/kitti_step_dataset/README.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+Author: TimoK93
+
+This project implements **`KITTI STEP Dataset`**
+
+### Dataset preparing
+
+After registration, the data images could be download from [KITTI-STEP](http://www.cvlibs.net/datasets/kitti/eval_step.php)
+
+You may need to follow the following structure for dataset preparation after downloading iSAID dataset.
```
Comment (xiexinch) [inline projects/kitti_step_dataset/README.md:15]:
```suggestion
You may need to follow the following structure for dataset preparation after downloading KITTI-STEP dataset.
```

**Review by xiexinch — APPROVED** [review]

**Review by csatsurnh — APPROVED** [review]

**Pull request merged**